File: tasks/implementations/python/tasks/web_scraper.py
from pathlib import Path
import re


def _extract(pattern: str, text: str):
    return re.findall(pattern, text, flags=re.IGNORECASE)


def run(size: str, fixtures_root: str) -> None:
    base = Path(fixtures_root) / "mock_site" / size
    index = (base / "index.html").read_text(encoding="utf-8")
    links = _extract(r'href="([^"]+)"', index)
    total = 0
    for idx, link in enumerate(links):
        text = (base / link).read_text(encoding="utf-8")
        title = _extract(r"<h2>(.*?)</h2>", text)[0]
        body = _extract(r"<p>(.*?)</p>", text)[0]
        total += (idx + 1) * (len(title) + len(body))
    print(total)

File: tasks/implementations/node/tasks/web_scraper.js
const fs = require('fs');
module.exports = function(size, fixturesRoot) {
  const base = `${fixturesRoot}/mock_site/${size}`;
  const index = fs.readFileSync(`${base}/index.html`, 'utf8');
  const links = [...index.matchAll(/href="([^"]+)"/g)].map((m) => m[1]);
  let total = 0;
  links.forEach((link, idx) => {
    const text = fs.readFileSync(`${base}/${link}`, 'utf8');
    const title = (text.match(/<h2>(.*?)<\/h2>/i) || [])[1] || '';
    const body = (text.match(/<p>(.*?)<\/p>/i) || [])[1] || '';
    total += (idx + 1) * (title.length + body.length);
  });
  console.log(total);
};
